tests/unit/discovery/py_info/test_py_info.py

```diff
@@ -378,6 +378,29 @@ def test_custom_venv_install_scheme_is_prefered(mocker):
     assert pyinfo.install_path("purelib").replace(os.sep, "/") == f"lib/python{pyver}/site-packages"
 
 
+@pytest.mark.skipif(not (os.name == "posix" and sys.version_info[:2] >= (3, 11)), reason="POSIX 3.11+ specific")
+def test_fallback_existent_system_executable(mocker):
+    current = PythonInfo()
+    # Posix may execute a "python" out of a venv but try to set the base_executable
+    # to "python" out of the system installation path. PEP 394 informs distributions
+    # that "python" is not required and the standard `make install` does not provide one
+
+    # Falsify some data to look like we're in a venv
+    current.prefix = current.exec_prefix = "/tmp/tmp.izZNCyINRj/venv"
```

Comment:
will this test fail if upstream accepts your PR?

Reply:
It shouldn't. I mock `sys. _base_executable` to return a nonexistent file to mimic the "broken" behavior so it should continue to work as expected even if the value `sys. _base_executable` would return is correct. Once upstream decides on the PR and whether to backport it, we can change the max version to which this quirk applies.